### DISCUSSION TITLE
Toggling between inputs

### DISCUSSION THREAD

**Discussion by dipsy-lot** [houmain/keymapper#189] [Q&A] [answered]
I love this program. There's so much customizability and I wonder if there's a way to press a set of buttons to make one input switch between different shortcuts

For example I want to be able to do this:

(AltLeft Q) >
Toggles between: ButtonLeft >> O
                           ButtonLeft >> ButtonLeft

Further explanation: Pressing alt+q will make left click become O and pressing it again will revert back to normal

**Comment by houmain** [comment, marked as answer]
Thanks! 
Of course, there are multiple ways. Something like:
```
AltLeft{Q} >> Virtual1
Virtual1{ButtonLeft} >> O
```
or
```
ModeSwitch = Virtual1
AltLeft{Q} >> ModeSwitch

[modifier = "ModeSwitch"]
ButtonLeft >> O

[default]
ButtonLeft >> ButtonLeft
```

**Reply by dipsy-lot** [reply]
Thanks it works 👍. I also wish to have some sort of indicator when it’s on and off. How would I go about that

Edit: If you want me to be specific, something like displaying on the corner of my screen “on” or “off”. But any type of indicator would help. 

Also, when using my left alt normally by itself, there's like a delay. Like it's waiting for another input but I want to be able to use left alt by itself with no delay.

**Reply by ristomatti** [reply]
You can extend the example above to have this:

```ini
[modifier = "ModeSwitch"]
ContextActive >> $(command-to-indicate-on-state) ^ $(command-to-indicate-off-state) 
ButtonLeft >> O
```

The `command-to-...` part is something you need to figure out yourself depending on your OS. I'm pretty sure there's some app for all operating systems that can display a tray icon or some type of a widget that can update it's state based on some external input like:
- executing a command with some parameters
- repeatedly read the contents of a text file

If you're familiar with shell scripting, you can also detect the `ModeSwitch` state externally within a script using the [`keymapperctl`](https://github.com/houmain/keymapper?tab=readme-ov-file#keymapperctl) command.
